client/azure/azure.go

```diff
@@ -0,0 +1,132 @@
+package azure
```

Comment:
To me having a client package that sets up the API connection makes sense. For k8s we extracted this to operatorkit. I don't think it makes sense to extract the Azure client at this stage.

I agree the business logic should be in services but I think those services can use the Azure client.